Some minor improvements for password plugin

matwey opened the pull request:
* Allow LDAP URI to communicate via UNIX socket (when available)
* Allow SASL bind for LDAP search queue.

matwey committed:
plugins/password/ldap_simple: Allow LDAP access using LDAP URI

Signed-off-by: Matwey V. Kornilov <matwey@sai.msu.ru>

alecpl commented:
Why sasl bind for search and not also for record modification? It would be good to describe new options in config.inc.php.dist. And it looks to me that 'password_ldap_uri' does not work in the 'ldap' driver. This needs to be fixed or noted.

matwey commented:
Thank you for the feedback!

- I think it is a good idea to enable SASL also for password update.
- `config.inc.php.dist` update is also on the way.
- Indeed, `ldap` driver uses deprecated library which doesn't support LDAP URI.

matwey committed:
plugins/password/ldap_simple: Allow LDAP SASL bind for admin

Signed-off-by: Matwey V. Kornilov <matwey@sai.msu.ru>

alecpl commented on plugins/password/config.inc.php.dist:
```diff
@@ -171,6 +171,13 @@ $config['password_ldap_host'] = 'localhost';
+// 'password_ldap_host' and 'password_ldap_port'.
+// Example: 'ldapi:///'
+// Default: null
+$config['password_ldap_uri'] = null;
```
Many other options e.g. `imap_host` support URI syntax, therefore I think it would make sense to support URIs in `password_ldap_host` instead.

matwey replied:
`password_ldap_host` is used instead of `password_ldap_uri`

alecpl commented on plugins/password/drivers/ldap_simple.php:
```diff
@@ -222,10 +243,28 @@ function search_userdn($rcmail, $ds)
-        if (!ldap_bind($ds, $search_user, $search_pass)) {
-            $this->_debug("S: ".ldap_error($ds));
-            return false;
+        switch ($rcmail->config->get('password_ldap_search_bind_method')) {
```
Do we really need a separate method/mech/realm for search? Is it a real use-case to have a different method for both binds?

matwey replied:
`password_ldap_search_`/`password_ldap_admin_` renamed to `password_ldap_`

matwey commented:
Sorry for this long delay. I am working on changes.

matwey committed:
plugins/password/ldap_simple: Allow LDAP SASL binding when search user DN

This may be particularly useful when UNIX socket connection is used,
or TLS certificate-based authentication is performed.

Signed-off-by: Matwey V. Kornilov <matwey@sai.msu.ru>
